kvcache/encoder.go

```diff
@@ -40,9 +43,25 @@ func NewEncoderCache() *EncoderCache {
 }
 
 func (c *EncoderCache) Init(backend ml.Backend, dtype ml.DType, capacity int32) {
```

Comment:
Should we have a `NewEncoderCache` now that we require initialization of `config`?